modules/admin/src/test/java/org/motechproject/admin/it/AdminBundleIT.java

```diff
@@ -114,6 +127,64 @@ public void testMessageController() throws IOException, InterruptedException {
         assertTrue("No messages listed", json.size() > 0);
     }
 
+    @Test
+    public void testUploadBundleFromRepository() throws IOException, InterruptedException {
+        boolean isUploadSuccessful = uploadBundle("Repository", "org.motechproject:cms-lite:LATEST", null,
+                "on","org.motechproject.cms-lite");
+
+        assertTrue(isUploadSuccessful);
+    }
+
+    @Test
+    public void testUploadBundleFromFile() throws IOException, InterruptedException {
+        File file = new File("src/test/resources/motech-upload-test-bundle.jar");
+
+        boolean isUploadSuccessful = uploadBundle("File", null, file, "on", "motech-upload-test-bundle");
+
+        assertTrue(isUploadSuccessful);
+    }
+
+    private boolean uploadBundle(String moduleSource, String moduleId, File bundleFile,
+                                 String startBundle, String bundleSymbolicName) throws IOException, InterruptedException {
+        String uri = String.format("http://%s:%d/admin/api/bundles/upload/", HOST, PORT);
+
+        HttpPost httpPost = new HttpPost(uri);
+        MultipartEntityBuilder entity = MultipartEntityBuilder.create();
+        Charset chars = Charset.forName("UTF-8");
+        entity.setCharset(chars);
+        entity.addTextBody("moduleSource", moduleSource, ContentType.MULTIPART_FORM_DATA);
+        if (moduleSource.equals("Repository")) {
+            entity.addTextBody("moduleId", moduleId, ContentType.MULTIPART_FORM_DATA);
+        }
+        else if (moduleSource.equals("File")) {
```

Comment:
The convention in MOTECH is to put else ifs and elses in the same line as the ending bracket of the previous block, after blank space, eg.  } else if {